ang/test/civiawards/directives/award.directive.spec.js

```diff
@@ -0,0 +1,255 @@
+              expect($scope.basicDetails.name).toBe('some_custom_text');
+            });
+          });
+        });
+      });
+
+      describe('when editing existing award', function () {
+        beforeEach(() => {
+          crmApi.and.returnValue($q.resolve({
+            caseType: AwardData,
+            additionalDetails: AwardAdditionalDetailsData
+          }));
+
+          createController({ ifNewAward: false });
+          $scope.$digest();
+        });
+
+        it('fetches the award information', () => {
+          expect(crmApi).toHaveBeenCalledWith({
+            caseType: ['CaseType', 'getsingle', { sequential: true, id: '10' }],
+            additionalDetails: ['AwardDetail', 'getsingle', { sequential: true, case_type_id: '10' }]
+          });
+        });
+
+        it('displays the basic awards information', () => {
+          expect($scope.basicDetails.title).toBe(AwardData.title);
+          expect($scope.basicDetails.name).toBe(AwardData.name);
+          expect($scope.basicDetails.description).toBe(AwardData.description);
+          expect($scope.basicDetails.isEnabled).toBe(AwardData.is_active === '1');
+        });
+
+        it('displays the additional awards information', () => {
+          expect($scope.awardDetailsID).toBe(AwardAdditionalDetailsData.id);
+          expect($scope.basicDetails.startDate).toBe(AwardAdditionalDetailsData.start_date);
+          expect($scope.basicDetails.endDate).toBe(AwardAdditionalDetailsData.end_date);
+          expect($scope.basicDetails.awardType).toBe(AwardAdditionalDetailsData.award_type);
+          expect($scope.basicDetails.awardManagers).toBe(AwardAdditionalDetailsData.award_manager.toString());
+        });
+
+        describe('name field', () => {
+          beforeEach(() => {
+            $scope.basicDetails.title = 'Some custom text';
+            $scope.$digest();
+          });
+
+          it('updating the title does not change the name field', () => {
+            expect($scope.basicDetails.name).toBe('new_award');
+          });
+        });
+      });
+    });
+
+    describe('when saving award', () => {
+      describe('when creating a new award', () => {
+        beforeEach(() => {
+          createController({ ifNewAward: true });
+          $scope.basicDetails.title = 'title';
+          $scope.basicDetails.description = 'description';
+          $scope.basicDetails.isEnabled = true;
+          $scope.basicDetails.selectedAwardStages = { 1: true };
+          $scope.basicDetails.awardManagers = '2,1';
+          $scope.basicDetails.startDate = AwardAdditionalDetailsData.start_date;
+          $scope.basicDetails.endDate = AwardAdditionalDetailsData.end_date;
+          $scope.basicDetails.awardType = AwardAdditionalDetailsData.award_type;
+
+          crmApi.and.returnValue($q.resolve({
+            values: [AwardAdditionalDetailsData]
+          }));
+          $scope.saveAward();
+        });
+
+        it('disables the save button initially', () => {
```

Comment:
`it momentarily disables the save button` (I'd even remove momentarily and/or initially)